internal/cmd/imagerunner/stop.go

```diff
@@ -0,0 +1,57 @@
+package imagerunner
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	cmds "github.com/saucelabs/saucectl/internal/cmd"
+	"github.com/saucelabs/saucectl/internal/http"
+	"github.com/saucelabs/saucectl/internal/segment"
+	"github.com/saucelabs/saucectl/internal/usage"
+	"github.com/spf13/cobra"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func StopCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "stop <runID>",
+		Short:        "Stop the running container",
+		SilenceUsage: true,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 || args[0] == "" {
+				return errors.New("no run ID specified")
+			}
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			err := http.CheckProxy()
+			if err != nil {
+				return fmt.Errorf("invalid HTTP_PROXY value")
+			}
+
+			tracker := segment.DefaultTracker
+
+			go func() {
+				tracker.Collect(
+					cases.Title(language.English).String(cmds.FullName(cmd)),
+					usage.Properties{}.SetFlags(cmd.Flags()),
+				)
+				_ = tracker.Close()
+			}()
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ID := args[0]
+			if err := imagerunnerClient.StopRun(context.Background(), ID); err != nil {
+				return fmt.Errorf("failed to stop the container: %v", err)
+			}
+
+			fmt.Printf("Stopping container %s...\n", ID)
```

Comment:
Shouldn't this line be logically logged before the API call happens? I don't know entirely whether the backend is blocking until the container actually has shut down, but if it does (and if it doesn't, that could change), the API will be blocking as well (and would be regardless if something's iffy with networking) and you'd only log this line afterwards. Meaning, the user will see no output whatsoever until this happens.

Reply:
The API returns 204 no content whatever I sent whether it should be successful or failed. So I don't think it can encounter any blocking. I even tried to close a non-existing container and it returned 204. I didn't see any error except I managed to let it encounter network issue.
So according to the behavior, I think it's an async operation and just sending a command.